crates/symbolicli/src/main.rs

```diff
@@ -0,0 +1,655 @@
+            ))
+            .unwrap();
+
+        tracing::info!(url = %attachments_url, "fetching attachments");
+
+        let response = client
+            .get(attachments_url.clone())
+            .send()
+            .await
+            .context("Failed to send request")?;
+
+        let attachments: Vec<Attachment> = if response.status().is_success() {
+            response
+                .json()
+                .await
+                .context("Failed to decode attachments")?
+        } else {
+            bail!(format!(
+                "Response from server: {}",
+                response
+                    .status()
+                    .canonical_reason()
+                    .unwrap_or("unknown error")
+            ));
+        };
+
+        let Some(minidump_id) = attachments
+            .iter()
+            .find(|attachment| attachment.r#type == "event.minidump")
+            .map(|attachment| &attachment.id) else {
+                return Ok(None);
+        };
+
+        let mut download_url = attachments_url.join(&format!("{minidump_id}/")).unwrap();
+        download_url.query_pairs_mut().append_pair("download", "1");
+
+        Ok(Some(download_url))
+    }
+
+    pub async fn download_minidump(
+        client: &reqwest::Client,
+        download_url: Url,
+    ) -> anyhow::Result<TempPath> {
+        tracing::info!(url = %download_url, "downloading minidump file");
+
+        let response = client
+            .get(download_url)
+            .send()
+            .await
+            .context("Failed to send request")?;
+
+        let minidump = if response.status().is_success() {
+            response
+                .bytes()
+                .await
+                .context("Failed to extract response body")?
+        } else {
+            bail!(format!(
+                "Response from server: {}",
+                response
+                    .status()
+                    .canonical_reason()
+                    .unwrap_or("unknown error")
+            ));
+        };
+
+        let mut temp_file = NamedTempFile::new().unwrap();
+        temp_file
+            .write_all(&minidump)
+            .context("Failed to write minidump to disk")?;
+
+        Ok(temp_file.into_temp_path())
+    }
+
+    pub async fn download_event(
+        client: &reqwest::Client,
+        base_url: &Url,
+        org: &str,
+        project: &str,
+        event_id: &str,
+    ) -> anyhow::Result<Event> {
+        let event_url = base_url
+            .join(&format!("projects/{org}/{project}/events/{event_id}/json/"))
+            .unwrap();
+
+        tracing::info!(url = %event_url, "downloading event json");
+        let response = client
+            .get(event_url.clone())
+            .send()
+            .await
+            .context("Failed to send request")?;
+
+        if response.status().is_success() {
+            response.json().await.context("Failed to decode event")
```

Comment:
I was wondering why there's no `?` at the end, but this is an explicit return already. Just a long one 😅